types/socket.io/index.d.ts

```diff
@@ -1,58 +1,41 @@
 // Type definitions for socket.io 1.4.4
 // Project: http://socket.io/
-// Definitions by: PROGRE <https://github.com/progre>, Damian Connolly <https://github.com/divillysausages>, Florent Poujol <https://github.com/florentpoujol>, KentarouTakeda <https://github.com/KentarouTakeda>, Alexey Snigirev <https://github.com/gigi>
+// Definitions by: PROGRE <https://github.com/progre>,
+//                 Damian Connolly <https://github.com/divillysausages>,
+//                 Florent Poujol <https://github.com/florentpoujol>,
+//                 KentarouTakeda <https://github.com/KentarouTakeda>, 
+//                 Alexey Snigirev <https://github.com/gigi>, 
+//                 Ayon Lee <https://github.com/hyurl>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 ///<reference types="node" />
 
-declare module 'socket.io' {
-	var server: SocketIOStatic;
-
-	export = server;
+declare module "socket.io" {
+	export = SocketIO;
 }
 
-interface SocketIOStatic {
-	/**
-	 * Default Server constructor
-	 */
-	(): SocketIO.Server;
-
-	/**
-	 * Creates a new Server
-	 * @param srv The HTTP server that we're going to bind to
-	 * @param opts An optional parameters object
-	 */
-	(srv: any, opts?: SocketIO.ServerOptions): SocketIO.Server;
-
-	/**
-	 * Creates a new Server
-	 * @param port A port to bind to, as a number, or a string
-	 * @param An optional parameters object
-	 */
-	(port: string|number, opts?: SocketIO.ServerOptions): SocketIO.Server;
-
-	/**
-	 * Creates a new Server
-	 * @param A parameters object
-	 */
-	(opts: SocketIO.ServerOptions): SocketIO.Server;
+declare function SocketIO(): SocketIO.Server;
```

Comment:
This will declare a global variable.
To make the types in `SocketIO` all you need to do (starting from the version in `master`) is to remove `declare module 'socket.io'` (moving its contents to the top-level) and rename `var server` to `declare const SocketIO: SocketIOStatic;`. Then the `const SocketIO` will merge with `namespace SocketIO` and you'll be able to use it like:
```ts
import socketIO = require('socket.io');
const x: socketIO.Server = ...;
```

Reply:
Thanks for the advice, but simply  rename `var server` to `declare const SocketIO: SocketIOStatic;` doesn't work, the editor tells that I cannot redefine the variable `SocketIO`, because the namespace in TypeScript is actually a variable that has already taken the name `SocketIO`. (but it works with functions). And it seems the `export = SocketIO` also works with `import... from ...` syntax since ts2.7. About that global variable thing, I've tried a little bit, no matter declare a function or a variable, it always seems like a global variable (in vscode), but actually not.

Reply:
> I cannot redefine the variable SocketIO, because the namespace in TypeScript is actually a variable

That is only a problem if the namespace contains something other than types. The version in master contains only types, so this should work if you start from the master branch. I've done this in #24261.

Reply:
Thanks for the hint: That is only a problem if the namespace contains something other than types.